clang/tools/clang-format/clang-format.el

```diff
@@ -216,17 +323,75 @@ uses the function `buffer-file-name'."
             (if incomplete-format
                 (message "(clang-format: incomplete (syntax errors)%s)" stderr)
               (message "(clang-format: success%s)" stderr))))
-      (delete-file temp-file)
+      (ignore-errors (delete-file temp-file))
       (when (buffer-name temp-buffer) (kill-buffer temp-buffer)))))
 
+
+;;;###autoload
+(defun clang-format-vc-diff (&optional style assume-file-name)
+  "The same as ‘clang-format-buffer’ but only operates on the vc
+diffs from HEAD in the buffer. If no STYLE is given uses
+‘clang-format-style’. Use ASSUME-FILE-NAME to locate a style config
+file. If no ASSUME-FILE-NAME is given uses the function
+‘buffer-file-name’."
+  (interactive)
+  (let ((tmpfile-vc-head nil)
+        (tmpfile-curbuf nil))
+    (unwind-protect
+        (progn
+          (setq tmpfile-vc-head
+                (make-temp-file "clang-format-vc-tmp-head-content"))
+          (clang-format--vc-diff-get-vc-head-file tmpfile-vc-head)
+          ;; Move the current buffer to a temporary file to take a
+          ;; diff. Even if current-buffer is backed by a file, we
+          ;; want to diff the buffer contents which might not be
+          ;; saved.
+          (setq tmpfile-curbuf (make-temp-file "clang-format-vc-tmp"))
```

Comment:
Suggest to pass this `tmpfile-curbuf` to `clang-format--vc-diff-get-vc-head-file`, then use `(:file FILE)` as the destination.
This will be faster and use less memory as it will do `process->buffer` instead of  `process->buffer->file`.

Reply:
Hmm? In `clang-format--vc-diff-get-vc-head-file` we do: ``(:file, tmpfile-vc-head)`, `tmpfile-curbuf` is the current contents of the buffer we are calling `clang-format-vc-diff` on.

Reply:
Ah, sorry - my mistake.